src/main/java/space/pxls/data/Database.java

```diff
@@ -1170,10 +1173,11 @@ public Integer createChatMessage(int authorID, long sent, String content, String
      * @param sent The chat message's creation epoch.
      * @param content The chat contents.
      * @param filtered The filtered chat contents.
+     * @param shadowBanned Whether or not the user sending the message is shadow-banned.
      * @return The new chat message's ID.
      */
-    public Integer createChatMessage(User author, long sent, String content, String filtered) {
-        return createChatMessage(author == null ? -1 : author.getId(), sent / 1000L, content, filtered);
+    public Integer createChatMessage(User author, long sent, String content, String filtered, boolean shadowBanned) {
+        return createChatMessage(author == null ? -1 : author.getId(), sent / 1000L, content, filtered, shadowBanned);
```

Comment:
```suggestion
        return createChatMessage(author == null ? -1 : author.getId(), sent / 1000L, content, filtered, author == null ? false : author.isShadowBanned());
```
(and removing the `boolean shadowBanned` argument)